frontend/src/modules/history/components/Translation.js

```diff
@@ -224,6 +224,12 @@ export class TranslationBase extends React.Component<InternalProps, State> {
         );
 
         let className = 'translation ' + this.getStatus();
+
+        if (!user || !user.isAuthenticated || isReadOnlyEditor) {
```

Comment:
Note that `isReadOnlyEditor` covers the `!user.isAuthenticated` case, so I think it can be the only condition here.